src/v/cluster/topics_frontend.cc

```diff
@@ -841,6 +843,27 @@ ss::future<topic_result> topics_frontend::do_create_topic(
     if (!units) {
         co_return make_error_result(assignable_config.cfg.tp_ns, units.error());
     }
+
```

Comment:
Above we have already called the allocator to allocate partition for this topic. Does the desctruction of the "units" object DTRT and "back out" that allocation?

Is there any reason we don't just do the topic check first? Then we don't even have to think about this because the topic check is non-mutating.

Reply:
Good point, will move this check towards the beginning of the function.